api/sirsidynix_authentication_provider.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+from gettext import gettext as _
+from typing import TYPE_CHECKING
+from urllib.parse import urljoin
+
+from api.authenticator import BasicAuthenticationProvider, PatronData
+from core.model.configuration import ExternalIntegration
+from core.util.http import HTTP
+
+if TYPE_CHECKING:
+    from requests import Response
+
+
+class SirsiDynixHorizonAuthenticationProvider(BasicAuthenticationProvider):
+    NAME = "SirsiDynix Horizon Authentication"
+    DESCRIPTION = "SirsiDynix Horizon Webservice Authentication"
+    FLOW_TYPE = "http://librarysimplified.org/authtype/sirsidynix-horizon"
+
+    class Keys:
+        APP_ID = "APP_ID"
+        CLIENT_ID = "CLIENT_ID"
+
+    SETTINGS = [
+        {
+            "key": ExternalIntegration.URL,
+            "label": _("Server URL"),
+            "description": _("The external server url."),
+            "required": True,
+            "default": "https://vendor1-sym.sirsidynix.net/ilsws_current/",
+        },
+        {
+            "key": Keys.APP_ID,
+            "label": _("Application ID"),
+            "description": _(
+                "This can be anything, it is used to group all requests to the API."
```

Comment:
I believe this should be a hard coded value, so all our CMs are sending the same thing.

Reply:
Hard Coded, or environment variable?

Reply:
I was thinking hard coded, but you are right, an environment variable is better idea.